src/nfa/delta.cc

```diff
@@ -331,17 +340,90 @@ void Delta::defragment(const BoolVector& is_staying, const std::vector<State>& r
             move->targets.rename(renaming);
         }
         p.erase(
-                std::remove_if(p.begin(), p.end(), [&](Move& move) -> bool {
+                std::remove_if(p.begin(), p.end(), [&](SymbolPost& move) -> bool {
                     return move.targets.empty();
                 }),
                 p.end()
         );
     }
 }
 
-const Post& Delta::operator[](State q) const {
-    if (q >= posts.size()) {
-        return empty_post;
+StatePost::moves_const_iterator::moves_const_iterator(const std::vector<SymbolPost>& symbol_posts, bool is_end)
```

Comment:
is_end strange, remove?

Reply:
m_ konvence?

Reply:
We will use `member_` convention here. Later, we may refactor all members to use the same convention.

Reply:
We will refactor the use of `is_end` later. Currently, it is the only way of setting the end iterator correctly. I am not sure how else would you set the end iterator in a cleaner way, in fact.

Reply:
We agreed on keeping the `is_end` flag to set the end iterator properly. The `member_` convention was applied elsewhere.